code/stats/medals.cpp

```diff
@@ -236,128 +256,158 @@ void parse_medal_tbl()
+			medal_stuff* medal_p;
+			medal_display_info display_t;
+			medal_display_info* display_p;
+			bool create_if_not_found = true;
 
 			required_string("$Name:");
-			stuff_string(temp_medal.name, F_NAME, NAME_LENGTH);
+			stuff_string(medal_t.name, F_NAME, NAME_LENGTH);
+
+			if (optional_string("+nocreate")) {
+				if (!Parsing_modular_table) {
+					Warning(LOCATION, "+nocreate flag used for medal in non-modular table\n");
+				}
+				create_if_not_found = false;
+			}
+
+			// Does this medal exist already?
+			// If so, load this new info into it
+			medal_p = get_medal_pointer(medal_t.name);
+			if (medal_p != nullptr) {
+				if (!Parsing_modular_table) {
+					error_display(1,
+						"Error:  Medal %s already exists.  All medal names must be unique.", medal_t.name);
+				}
+				display_p = &Medal_display_info[get_medal_position(medal_p->name)];
+			} else {
+				// Don't create poof if it has +nocreate and is in a modular table.
+				if (!create_if_not_found && Parsing_modular_table) {
+					if (!skip_to_start_of_string_either("$Name:", "#end")) {
+						error_display(1, "Missing [#end] or [$Name] after medal %s", medal_t.name);
+					}
+					continue;
+				}
 
-			// is this rank?  if so, save it
-			if (!stricmp(temp_medal.name, "Rank"))
-				Rank_medal_index = Num_medals;
+				Medals.push_back(medal_t);
+				medal_p = &Medals[Medals.size() - 1];
+				Medal_display_info.push_back(display_t);
+				display_p = &Medal_display_info[Medal_display_info.size() - 1];
+			}
 
 			if (optional_string("$Alt Name:")) {
-				stuff_string(temp_medal.alt_name, F_NAME);
+				stuff_string(medal_p->alt_name, F_NAME);
 			}
 
-			required_string("$Bitmap:");
-			stuff_string(temp_medal.bitmap, F_NAME, MAX_FILENAME_LEN);
+			if (optional_string("$Bitmap:")) {
+				stuff_string(medal_p->bitmap, F_NAME, MAX_FILENAME_LEN);
+			}
+
+			// Check here that the medal has a bitmap. If not, then error out
+			if (!stricmp(medal_p->bitmap, "")) {
+				error_display(1, "Missing valid bitmap file for medal %s", medal_p->name);
+			}
 
 			if (optional_string("+Position 640:")) {
-				stuff_int(&temp_display.coords[GR_640].x);
-				stuff_int(&temp_display.coords[GR_640].y);
+				stuff_int(&display_p->coords[GR_640].x);
+				stuff_int(&display_p->coords[GR_640].y);
 			}
-			else if (Num_medals < NUM_MEDALS_FS2) {
-				temp_display.coords[GR_640].x = Default_medal_coords[GR_640][Num_medals][0];
-				temp_display.coords[GR_640].y = Default_medal_coords[GR_640][Num_medals][1];
+			else if (Medals.size() < NUM_MEDALS_FS2) {
+				display_p->coords[GR_640].x = Default_medal_coords[GR_640][Medals.size()][0];
+				display_p->coords[GR_640].y = Default_medal_coords[GR_640][Medals.size()][1];
 			}
 			else {
-				Warning(LOCATION, "No default GR_640 position for medal '%s'!", temp_medal.name);
-				temp_display.coords[GR_640].x = 0;
-				temp_display.coords[GR_640].y = 0;
+				Warning(LOCATION, "No default GR_640 position for medal '%s'!", medal_p->name);
```

Comment:
Since this is during table parsing, ``error_display(0, "");`` is preferable to Warning